### DISCUSSION TITLE
Retrieving repository total disk usage

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [borgbackup/borg#6509] [Q&A] [answered]
Is there a built-in way to retrieve the size of a repository + cache size on disk **from the client**?

The repository is not on the same machine as the client, and `borg serve` is used as forced command at the moment, so retrieving the disk usage directly is not preferred.

I did find that `borg info` contains so-called 'cache stats'. But those are stats for the Borg [files cache](https://borgbackup.readthedocs.io/en/stable/internals/data-structures.html#cache).

I'd like to know:

* The size of the files cache for the specific repository. I know how to get that information, see above.
* The size of the repository on disk
* ... more? Are repository-specific files saved somewhere else?

**Comment by ThomasWaldmann** [comment, marked as answer]
IIRC, the overall repo disk usage (including all overheads, like e.g. for the repo index) is not available via the rpc interface.

But, if you enable quotas for a repo, it will log the quota usage at INFO level at commit time (which does **not** include some overheads).

**Reply by WilliamDEdwards** [reply]
In that case, I will look at the size of the actual directory after all.

Are there any directories aside from the files cache and the repository itself to take into account?

Thank you for your response.

**Reply by ThomasWaldmann** [reply]
there is:
- the repo directory
- ~/.cache/borg/... (usual location, can be redirected via env var)
- ~/.config/borg/... (usual location, can be redirected via env var)
- temp directory

**Reply by WilliamDEdwards** [reply]
For future readers: one thing that should be noted is that the [cache is local](https://borgbackup.readthedocs.io/en/stable/faq.html#can-i-backup-from-multiple-servers-into-a-single-repository). I.e. it's not necessarily relevant to calculate the size of the repository on server, if that's what you're interested in.
